week-8-practise/1-code-refactoring.js

```diff
@@ -1,35 +1,27 @@
 /* EXERCISE 1 - Code Refactoring
- * This script functions as a library. 
+ * This script functions as a library.
  * Its purpose is to provide a set of functions that users can use to manipulate and decorate text.
  *
  * Using the coding practices you have learnt (you can use the syllabus as reference),
- * edit the code so that it establishes good coding standards. 
+ * edit the code so that it establishes good coding standards.
  *
  * There may be both syntax and logical errors present.
  */
 
-var reversed;
-let capitilised;
-
 // Function that reverses and capitlises text
-exports function reverseAndCaptilise(text) 
-  var capitlised = text.toUpper()
-  reversed = capitilised.split('').reverse().join('') 
-
-  return reversed
-  }
+export function reverseAndCaptilise(text) {
+  return text
+    .split("")
+    .reverse()
+    .join("")
+    .toUpperCase();
+}
 
 // Function that returns text with every other letter capitlised
 export function everyOtherLetterCapitilised(text) {
-  for (i==0; i++; i<text.length) {
-      if (i/2==0) {
-       text[i] = text[i].toUpperCase() 
-    }    
-  } 
+  return text.toUpperCase();
```

Comment:
The intention of the function is to make every other letter captilised (i.e. if input is 'practise', output is 'PrAcTiSe'), so the 'for' loop should stay.

However, there is a mistake in the for loop. What about the name of the function?

Reply:
Thanks @sansaid  for your review I have updated and refactor the code again, the name of the function is descriptive but long.